frontend/dockerfile/linter/ruleset.go

```diff
@@ -55,32 +55,32 @@ var (
 	}
 	RuleReservedStageName = LinterRule[func(string) string]{
 		Name:        "ReservedStageName",
-		Description: "Reserved stage names should not be used to name a stage",
+		Description: "Reserved words should not be used as stage names",
 		URL:         "https://docs.docker.com/go/dockerfile/rule/reserved-stage-name/",
 		Format: func(reservedStageName string) string {
 			return fmt.Sprintf("Stage name should not use the same name as reserved stage %q", reservedStageName)
 		},
 	}
 	RuleJSONArgsRecommended = LinterRule[func(instructionName string) string]{
 		Name:        "JSONArgsRecommended",
-		Description: "JSON arguments recommended for ENTRYPOINT/CMD to prevent unintended behavior related to OS signals",
+		Description: "JSON arguments recommended for ENTRYPOINT/CMD to prevent unintended behavior related to OS signals, unless using a custom shell",
```

Comment:
nit: "unless using a custom shell" is not needed here. If the user knows what "custom shell" is then they will not see this warning. Otherwise it will just confuse them.

Explaining the "custom shell" behavior is the linked URL is good.